base/shell/explorer/trayprop.cpp

```diff
@@ -270,13 +270,47 @@ class CStartMenuSettingsPage : public CPropertyPageImpl<CStartMenuSettingsPage>
     }
 };
 
+WORD *TEMPLATE_ExtractRect(LPARAM lParam, RECT &rc)
+{
+    WORD *ptr = reinterpret_cast<WORD*>(lParam);
+    ((*ptr == 1) && (*(ptr + 1) == 0xffff)) ? ptr += 9 : ptr += 5;
+    rc.left = *ptr;
+    rc.top = *(ptr + 1);
+    rc.right = *(ptr + 2);
+    rc.bottom = *(ptr + 3);
+    return ptr;
+}
+
+#define EXTRA_BUTTON_SPACE 20
+
 static int CALLBACK
 PropSheetProc(HWND hwndDlg, UINT uMsg, LPARAM lParam)
 {
     // NOTE: This callback is needed to set large icon correctly.
     HICON hIcon;
     switch (uMsg)
     {
+        case PSCB_PRECREATE:
+        {
+            RECT rc;
+            HRSRC hRes;
+            LPVOID Template;
+
+            if ((hRes = FindResource(NULL, MAKEINTRESOURCE(IDD_TASKBARPROP_TASKBAR),
+                                     (LPWSTR)RT_DIALOG)) != NULL)
+                Template = LoadResource(NULL, hRes);
+            else 
+                return 0;
+
+            WORD *temple = TEMPLATE_ExtractRect(lParam, rc);
+            TEMPLATE_ExtractRect((LPARAM)Template, rc);
+            AdjustWindowRectEx(&rc, WS_CAPTION, 0, WS_EX_PALETTEWINDOW);
```

Comment:
Dialog units do not 1:1 translate to pixels.
The conversion is based on the device and the font.
You could use `MapDialogRect`.
See: https://support.microsoft.com/en-us/help/125681/how-to-calculate-dialog-base-units-with-non-system-based-font

But it would probably also work (and be easier) to just create the dialog, and when it's created move it?